### PR TITLE
fix: Crash on objects that are null or have unmappable values

### LINKED ISSUE #35: Deep comparison with dates can fail
When comparing arrays of objects where some values in the objects are dates can result in a failed comparison. In particular, even if the dates are different deepEqualInAnyOrder will report that the objects are the same. 

Here is an example from runkit: 

https://runkit.com/embed/udx9x7ubw5xd

The deepEqualInAnyOrder passes but the have.deep.members assertion fails (which is the correct thing to do)

### PR DESCRIPTION
**Pull request by kf6kjg**
Likely fixes #35, but I'm not certain.  I have been using this patch in my projects for some time and am finally getting around to sending it back upstream.

### CODE REVIEW

**Comment by kf6kjg** [comment]
Build fail is on the use of `catch` without a parameter, aka optional catch binding.   The syntax is valid as of [Node 10.3](https://node.green/#ES2019-misc-optional-catch-binding), and has [reasonable browser support](https://caniuse.com/mdn-javascript_statements_try_catch_optional_catch_binding).  Bypassing that feature for greater compat is possible, but isn't so clean.

**Comment by oprogramador** [comment]
Thanks for this PR.

I probably noticed it soon after opening but I didn't have time to fix the failing build.

Sorry for no reply from my part.

Maybe I'll fix the build or I'll open my own PR fixing that issue but I'm not sure whether I'll find time for that.

File: src/index.js [hunk 1]
```diff
@@ -3,11 +3,16 @@ import sortAny from 'sort-any';
 
 const sortDeep = (object) => {
   if (!Array.isArray(object)) {
-    if (!(typeof object === 'object') || object === null) {
+    // == null checks for undefined as well as null.
+    if (object == null || (typeof object !== 'object')) {
```
Comment (oprogramador) [inline src/index.js:7]: for `undefined`, `typeof object === 'object'` is false

Comment (kf6kjg) [reply src/index.js:7]: Quite right. Didn't see that then. Would you like me to revert this line?

Comment (oprogramador) [reply src/index.js:7]: no, thanks, I'm working on fixing this issues, with prodiving some tests

**Comment by oprogramador** [comment]
@kf6kjg 
I fixed the #35 issue (about dates) in the `1.1.13` version.

Which other data types do you need?
Maybe Map, WeakMap, Error?

**Comment by kf6kjg** [comment]
Not had such a need as yet. Thanks for the fixes! :)

**Pull request closed without merging**